### ISSUE TITLE
Solvability in mean zero spaces

### ISSUE THREAD

**Issue by dalbrit** [gridap/Gridap.jl#743]
In the following simple Poisson solver in 2d with periodic BCs, I expect to use a zero mean space. The implementation below, without the zero mean constraint, works; when I add in constraint=:zeromean, everything compiles up until the solve. Any ideas?

```
using Gridap

model = CartesianDiscreteModel((0,2π,0,2π),(100,100);isperiodic=(true,true))
order = 2

# This works although I did not put it in a mean zero function space. Why?
V = TestFESpace(model,ReferenceFE(lagrangian,Float64,order);conformity=:H1) #,constraint=:zeromean)
# When I use the command below instead, it won't solve.
# V = TestFESpace(model,ReferenceFE(lagrangian,Float64,order);conformity=:H1),constraint=:zeromean)
U = TrialFESpace(V)

degree = order
Ω = Triangulation(model)
dΩ = Measure(Ω,degree) 

f(x) = cos(x[1])+sin(x[2]) # This is mean zero.
a(u,v) = ∫( ∇(v)⋅∇(u) )*dΩ
b(v) = ∫( v*f )*dΩ

op = AffineFEOperator(a,b,U,V)
ls = LUSolver()
solver = LinearFESolver(ls)
uh = solve(solver,op)
writevtk(Ω,"results",cellfields=["uh"=>uh])
```

**Comment by fverdugo** [comment]
I can reproduce the error. It's a bug. Fortunately, this one is easy to fix (WIP).

**Comment by fverdugo** [comment]
@dalbrit  The fix is coming in PR #746

>  This works although I did not put it in a mean zero function space. Why?

In some situations, it is possible to solve a system with a singular matrix if one provides an initial guess to the linear solver. In other cases, the linear solver is able to deal with a singular matrix and compute a least squares solution. (I have not checked what is playing out in this particular case).

> degree = order

BTW, I would use a higher integration degree (e.g. 2*order) to better integrate the rhs.
